consumers/notifiers/service.go

```diff
@@ -4,45 +4,177 @@
 
-// Service reprents a notification service.
+// Service represents a notification service.
 type Service interface {
+	// CreateNotifiers creates notifiers for certain group identified by the provided ID
+	CreateNotifiers(ctx context.Context, token string, notifiers ...things.Notifier) ([]things.Notifier, error)
+
+	// ListNotifiersByGroup retrieves data about a subset of notifiers
+	// related to a certain group identified by the provided ID.
+	ListNotifiersByGroup(ctx context.Context, token string, groupID string) ([]things.Notifier, error)
+
+	// ViewNotifier retrieves data about the notifier identified with the provided ID
+	ViewNotifier(ctx context.Context, token, id string) (things.Notifier, error)
+
+	// UpdateNotifier updates the notifier identified by the provided ID, that
+	// belongs to the user identified by the provided key.
+	UpdateNotifier(ctx context.Context, token string, notifier things.Notifier) error
+
+	// RemoveNotifiers removes the notifiers identified with the provided IDs, that
+	// belongs to the user identified by the provided key.
+	RemoveNotifiers(ctx context.Context, token, groupID string, id ...string) error
+
 	consumers.Consumer
 }
 
 var _ Service = (*notifierService)(nil)
 
 type notifierService struct {
-	auth     mainflux.AuthServiceClient
-	idp      mainflux.IDProvider
-	notifier Notifier
-	from     string
+	auth         mainflux.AuthServiceClient
+	idp          mainflux.IDProvider
+	notifier     Notifier
+	from         string
+	notifierRepo NotifierRepository
+	things       mainflux.ThingsServiceClient
 }
 
 // New instantiates the subscriptions service implementation.
-func New(auth mainflux.AuthServiceClient, idp mainflux.IDProvider, notifier Notifier, from string) Service {
+func New(auth mainflux.AuthServiceClient, idp mainflux.IDProvider, notifier Notifier, from string, notifierRepo NotifierRepository, things mainflux.ThingsServiceClient) Service {
 	return &notifierService{
-		auth:     auth,
-		idp:      idp,
-		notifier: notifier,
-		from:     from,
+		auth:         auth,
+		idp:          idp,
+		notifier:     notifier,
+		from:         from,
+		notifierRepo: notifierRepo,
+		things:       things,
 	}
 }
 
 func (ns *notifierService) Consume(message interface{}) error {
+	ctx := context.Background()
+
 	msg, ok := message.(messaging.Message)
 	if !ok {
 		return errors.ErrMessage
 	}
 
-	err := ns.notifier.Notify(ns.from, msg.Profile.Notifier.Contacts, msg)
+	if msg.Profile.SmtpID == "" && msg.Profile.SmppID == "" {
+		return apiutil.ErrMissingID
+	}
+
+	if msg.Profile.SmtpID != "" {
+		smtp, err := ns.notifierRepo.RetrieveByID(ctx, msg.Profile.SmtpID)
+		err = ns.notifier.Notify(ns.from, smtp.Contacts, msg)
+		if err != nil {
+			return errors.Wrap(ErrNotify, err)
+		}
+	}
+
+	if msg.Profile.SmppID != "" {
+		smpp, err := ns.notifierRepo.RetrieveByID(ctx, msg.Profile.SmppID)
+		err = ns.notifier.Notify(ns.from, smpp.Contacts, msg)
+		if err != nil {
+			return errors.Wrap(ErrNotify, err)
+		}
+	}
```

Comment:
Do we have to check this here? Conumers should consume only messages published in the proper NATS topic.